pkg/build/pipelines/git-checkout.yaml

```diff
@@ -24,7 +24,36 @@ inputs:
     description: |
       The branch to check out, otherwise HEAD is checked out.
 
+  tag:
+    description: |
+      The tag to check out.
+  
+  expected-commit:
+    description: |
+      The expected commit hash
+
 pipeline:
   - runs: |
-      [ -n '${{inputs.branch}}' ] && branch='--branch ${{inputs.branch}}'
-      git clone $branch --depth ${{inputs.depth}} ${{inputs.repository}} ${{inputs.destination}}
+      if [ -z "${{inputs.branch}}" ] && [ -z "${{inputs.tag}}" ]; then
+        echo "Warning (git-checkout): you have not specified a branch or tag."
+      fi
+
+      [ -n '${{inputs.branch}}' ] && clone_target='--branch ${{inputs.branch}}'
+      [ -n '${{inputs.tag}}' ] && clone_target='tags/${{inputs.tag}}'
+
+      git clone $clone_target --depth ${{inputs.depth}} ${{inputs.repository}} ${{inputs.destination}}
+
+      cd ${{inputs.destination}}
+
+      if [ -z "${{inputs.expected-commit}}" ]; then
+        echo "Warning (git-checkout): no expected-commit"
+      else
+        [ -n '${{inputs.branch}}' ] && remote_commit=$(git rev-list -1 origin/"${{inputs.branch}}")
```

Comment:
Does it make sense to specify an expected commit when you specify a branch? Those are more or less expected to move as I understand it.

Maybe this should fail if you specify `branch`+`expected-commit`, and warn if you specify `tag` _without_ `expected-commit`, and verify when you specify `tag`+`expected-commit`.

What do you think?

Reply:
even though we know branch is a moving target, i kind of want to warn here because in the end, what we want is reproduciblility right? having a pinpoint git commit sha is still better if we use this pipeline in Wolfi.